exporter/containerimage/export.go

```diff
@@ -258,7 +258,7 @@ func (e *imageExporterInstance) Export(ctx context.Context, src *exporter.Source
 				}
 				tagDone(nil)
 
-				if src.Ref != nil && e.unpack {
```

Comment:
This `src.Ref != nil` would have never been a case for Dockerfile anyway, right?

Reply:
In the dockerfile frontend we call `SetRef` for single-platform, and `AddRef` for multi-platform:

https://github.com/moby/buildkit/blob/5b9a9ce7df4fe1d70ada3cc899c2b223632d3e68/frontend/dockerui/build.go#L67-L73

So in the single-platform case `src.Ref != nil`, in the multi-platform case `src.Refs != nil`.

Reply:
Then it could be a breaking change for non-multi platform. But I think logically, only strictly relying on `.unpack` is correct.

Reply:
See https://github.com/moby/buildkit/pull/3982: we used to rely on `unpack` only, prior to https://github.com/moby/buildkit/pull/3251.

Reply:
Ah, sorry, I read this as "always try to unpack" when not multi-platform, but that is not at all what the previous code did.